vllm/engine/arg_utils.py

```diff
@@ -364,8 +364,9 @@ class EngineArgs:
     calculate_kv_scales: bool = CacheConfig.calculate_kv_scales
 
     additional_config: Optional[Dict[str, Any]] = None
-    enable_reasoning: Optional[bool] = None
-    reasoning_parser: Optional[str] = DecodingConfig.reasoning_backend
+    enable_reasoning: Optional[bool] = None  # DEPRECATED
```

Comment:
Actually this will means users who use EngineArgs will fail

let's make this into a property and set their value based on reasoning parser, and add a `@deprecated` decorator. wdyt?

cc @chaunceyjiang

Reply:
Actually I don't think this would change a lot, because b4 users will have to specify both `enable_reasoning` and `reasoning_parser` regardless.

It should be ok then

Reply:
> Actually this will means users who use EngineArgs will fail

I'm not quite getting it. The field isn't actually removed right now. When users use `EngineArgs`, it doesn't throw an error .They can still use it like this.
```
EngineArgs(enable_reasoning=True,reasoning_parser="qwen3")
```

Reply:
yep, please disregard this message. I was thinking about a separate behaviour where `--reasoning-parser` implies `--enable-reasoning`